src/varity/ref_gene.clj

```diff
@@ -200,15 +297,56 @@
                   :tx-end (:end transcript)
                   :strand strand))))
 
+(defn- ->genomic-gff-region
+  [feature-map [transcript-id transcript]]
+  (let [exons (->> (get-in feature-map [:exon transcript-id])
+                   (filter :exon-number)
+                   (sort-by :exon-number))
+        cdss (->> (get-in feature-map [:cds transcript-id])
+                  (filter :cds-number)
+                  (sort-by :cds-number))
+        [exons cdss strand] (case (:strand transcript)
+                              "+" [exons cdss :forward]
+                              "-" [(reverse exons) (reverse cdss) :reverse])]
+    (cond-> transcript
+      (seq exons) (assoc :exon-count (count exons)
+                         :exon-start (:start (first exons))
+                         :exon-end (:end (last exons))
+                         :exon-ranges (mapv #(vector (:start %) (:end %)) exons)
+                         :exon-frames (mapv :frame exons))
+      (empty? exons) (assoc :exon-count 0
+                            :exon-start nil
+                            :exon-end nil
+                            :exon-ranges []
+                            :exon-frames [])
+      (seq cdss) (assoc :cds-start (apply min (map :start cdss))
+                        :cds-end (apply max (map :end cdss)))
+      (empty? cdss) (assoc :cds-start (:start transcript)
+                           :cds-end (:end transcript))
+
+      true (assoc :tx-start (:start transcript)
+                  :tx-end (:end transcript)
+                  :strand strand))))
+
 (defn load-gencode
   [f parse-line & {:keys [chunk-size] :or {chunk-size 10000}}]
   (with-open [rdr (io/reader (util/compressor-input-stream f))]
     (let [feature-map (->> (line-seq rdr)
                            (partition-all chunk-size)
                            (pmap #(doall (keep parse-line %)))
                            (apply concat)
-                           build-feature-map)]
-      (doall (map (partial ->region feature-map) (:transcript feature-map))))))
+                           build-gencode-feature-map)]
+      (doall (map (partial ->gencode-region feature-map) (:transcript feature-map))))))
+
+(defn load-genomic
```

Comment:
This function's definition is redundant because `load-gencode` does almost the same things and I think it stems from my fault when implement `load-gencode`.
I should have implemented `load-gencode` depends on `load-gtf` and `load-gff3`, but I had some misunderstandings about the project name (GENCODE) and the file formats (latter is more general).

Can you re-implemt these functions depending on each other like below?

```mermaid
flowchart TD
    load-gencode --> load-gff3
    load-gencode --> load-gtf3
    load-ncbi-genome-annoation --> load-gtf3
```

IMO, parsing gff3 and gtf is parameterrized by  `parse-line` and `parse-attributes` and the latter is the part of the former.